cli/internal/core/engine.go

```diff
@@ -422,3 +431,45 @@ func (e *Engine) GetTaskGraphDescendants(taskID string) ([]string, error) {
 	sort.Strings(stringDescendents)
 	return stringDescendents, nil
 }
+
+func (e *Engine) getPipelineFromWorkspace(workspaceName string) (*fs.Pipeline, error) {
+	cachedPipeline, ok := e.pipelines[workspaceName]
+	if ok {
+		return cachedPipeline, nil
+	}
+
+	// TODO(mehulkar) what is this value when it's a non-root workspace?
+	dir := e.completeGraph.WorkspaceInfos[workspaceName].Dir
+
+	var pkgJSON *fs.PackageJSON
+	repoRoot := e.completeGraph.RepoRoot
+
+	// dirAbsolute := turbopath.AbsoluteSystemPath(dir)
+	dirAbsolutePath := repoRoot.UntypedJoin(dir.ToString())
+
+	// We need to get a package.json, because turbo.json can "synthesize" tasks from it
+	// for single-package repos.
```

Comment:
Does the root `package.json` still need to be read for monorepos?

Reply:
Yes, but only because it logs a warning when it sees a `turbo` key in there (which was the old way to define the pipeline). we've removed support for that already, but there's still a warning. I'd like to drop that warning eventually, and when that happens we can have two methods, one that synthesizes (and needs package.json), and one that doesn't